### ISSUE TITLE
git-cliff musl version is very slow compared to shared lib one

### ISSUE THREAD

**Issue by inglor** [orhun/git-cliff#630]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

The musl verison of the application is slow compared to the one using glibc one.

### Steps To Reproduce

Execute git-cliff musl and observe being slow

### Expected behavior

Same performance as glibc

### Software information

- Operating system: Amazon Linux 2
- Rust version: N/a (downloaded the provided binary)
- Project version: latest

**Comment by orhun** [comment]
I don't have much experience with MUSL but I ran a profiler for comparison:

- GNU: <https://share.firefox.dev/3UnZtZX> (336ms)
- MUSL: <https://share.firefox.dev/4a04i1h> (407ms)

There isn't a big difference here. It might be a good idea to investigate the calls though.

Can you share the repo you are running `git-cliff` with?

**Comment by inglor** [comment]
Unfortunately no since it's internal at work. But can run profiler myself potentially.

**Comment by orhun** [comment]
Sounds good, let me know.

**Comment by orhun** [comment]
Here are some instructions:

```sh
git clone https://github.com/orhun/git-cliff && cd git-cliff/

cargo build
samply record target/debug/git-cliff --no-exec -w "$workdir"

rustup target add x86_64-unknown-linux-musl
cargo build --target x86_64-unknown-linux-musl
samply record target/x86_64-unknown-linux-musl/debug/git-cliff --no-exec -w "$workdir"
```

Then on the browser you can _upload_ `profile.json` at top right.

**Comment by orhun** [comment]
Had any chance to try out the profiler?

**Comment by orhun** [comment]
Yeah, that might be the case. Can you share some execution times? I'm just curious about how slow it is.

Also there is this blog post from 2020 FWIW: <https://andygrove.io/2020/05/why-musl-extremely-slow/>
